File: web/apps/console/src/shared/api/src/types/ThemeInstallationResponse.ts
/**
 * ThemeInstallationResponse — wire shape returned by
 * GET /api/v1/branding/theme (200) and PUT /api/v1/branding/theme (200).
 *
 * Hand-written to mirror what `bun run generate` would emit from the
 * Plexor.Host openapi.json document. The kubb regeneration would
 * overwrite this file — keep the shape in lock-step with the
 * C# `ThemeInstallationResponse` class so a future regen doesn't
 * break the FE bindings.
 */
/* eslint-disable */
// @ts-nocheck

export type ThemeInstallationResponse = {
  /**
   * @type string, uuid
   */
  orgId: string;
  /**
   * @type string
   */
  themeId: string;
  /**
   * @type string
   */
  name: string;
  /**
   * @type string
   */
  version: string;
  /**
   * @type string
   */
  author: string;
  /**
   * @description Hex-encoded HMAC-SHA256 persisted on the row.
   * @type string
   */
  manifestSignature: string;
  /**
   * @type string, date-time
   */
  activatedAt: string;
  /**
   * @type string, uuid
   */
  activatedBy: string;
};

File: src/modules/Plexor.Modules.Branding/Plexor.Modules.Branding.Api/Models/Responses/ThemeInstallationResponse.cs
// SPDX-License-Identifier: Apache-2.0
// ============================================================================
// ThemeInstallationResponse — wire shape for the GET /
// PUT /api/v1/branding/theme response. Initialised-class shape
// matching the rest of the Branding API surface.
// ============================================================================

namespace Plexor.Modules.Branding.Api.Models.Responses;

/// <summary>
///     Wire shape for the marketplace installation row. Returned
///     by <c>GET /api/v1/branding/theme</c> (200) and
///     <c>PUT /api/v1/branding/theme</c> (200). 404 when no row
///     has been activated yet — the controller maps that to a
///     separate ProblemDetails shape.
/// </summary>
public sealed class ThemeInstallationResponse
{
    /// <summary>Tenant scope — the org whose installation this
    /// row records.</summary>
    public Guid OrgId { get; init; }

    /// <summary>Stable marketplace id of the installed theme.</summary>
    public string ThemeId { get; init; } = string.Empty;

    /// <summary>Publisher-friendly display name
    /// (<c>"Synthwave Night — Dark"</c>).</summary>
    public string Name { get; init; } = string.Empty;

    /// <summary>Publisher-supplied SemVer string.</summary>
    public string Version { get; init; } = string.Empty;

    /// <summary>Publisher handle or org name
    /// (<c>"plexor-themes"</c>).</summary>
    public string Author { get; init; } = string.Empty;

    /// <summary>Hex-encoded HMAC-SHA256 the publisher attached
    /// to the canonical manifest bytes at install time. The FE
    /// boot script can re-verify the same signature against
    /// <c>window.__PLEXOR_CONFIG__</c> on every reload.</summary>
    public string ManifestSignature { get; init; } = string.Empty;

    /// <summary>UTC time the activation was applied.</summary>
    public DateTimeOffset ActivatedAt { get; init; }

    /// <summary>Id of the user that applied the activation.</summary>
    public Guid ActivatedBy { get; init; }
}
